kratos/utilities/delaunator_utilities.cpp

```diff
@@ -234,6 +234,77 @@ std::vector<std::size_t> ComputeTrianglesConnectivity(const std::vector<Point>&
     return connectivities;
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::pair<std::vector<std::size_t>, std::vector<double>> ComputeTrianglesConnectivity(
+    const std::vector<double>& rCoordinates,
+    const std::vector<std::array<double,2>>& rSegments,
+    const double AreaConstraint)
+{
+    // Creating the containers for the input and output
+    struct triangulateio in_mid, out_mid, vorout_mid;
+
+    InitializeTriangulateIO(in_mid);
+    InitializeTriangulateIO(out_mid);
+    InitializeTriangulateIO(vorout_mid);
+
+    // Initialize the boundary points coordinates list
+    in_mid.numberofpoints = rCoordinates.size()/2;
+    in_mid.pointlist = (REAL *) malloc(in_mid.numberofpoints * 2 * sizeof(REAL));
```

Comment:
you never `free` any of the memory you allocate
also why using the C-way?

Reply:
The allocated memory is freed at the end using the CleanTriangulateIO function

Reply:
hm I would say this is bad design. Usually the one allocating the memory also frees it. I had quite a hard time with this in the recent past and was taught by @pooyan-dadvand to do it like that

Then also what does `InitializeTriangulateIO` do?

Reply:
`InitializeTriangulateIO` initializes the whole variablelist needed in the library with zeros and nullptr. This is done for the input, output and voroutput. Depending on the switches used some of the variables will be ignored, they still have to initialized. The `CleanTriangulateIO` frees the used memory completely. I hope I made it a little bit clearer. I get your point that the one allocating the memory should also free it. However, I think it would be quite cumbersome to free the complete variable list manually.

Reply:
By doing it that way we reuse the already existent `InitializeTriangulateIO` and `CleanTriangulateIO` to allocate/deallocate. The idea of this PR was to enable the constrained Delaunay reusing as much as possible the previous work of @loumalouomega . 

In any case I don't think it is a bad design at all as we are just simply "hiding" de allocation and deallocation instructions in two auxiliary functions.

Reply:
> By doing it that way we reuse the already existent InitializeTriangulateIO and CleanTriangulateIO to allocate/deallocate. 

but why do you then still allocate memory manually?

> In any case I don't think it is a bad design at all as we are just simply "hiding" de allocation and deallocation instructions in two auxiliary functions.

this makes perfect sense but it is not clear from reading the code! Hence my comments/confusion. I see `new` or `malloc` but not the corresponding `delete`/`free`. Also the function name doesn't make this clear.
IMO you could improve the naming of the methods to make clear that they do sth with memory as this is a sensitive thing

In any case I am only suggesting, maybe I don't see all the points

Only once I activate valgrind in the CI I might come back and knock at your door ;) :D

Reply:
> > By doing it that way we reuse the already existent InitializeTriangulateIO and CleanTriangulateIO to allocate/deallocate.
> 
> but why do you then still allocate memory manually?
> 

Because in the standard Delaunay this is not needed. As we require it for the constrained Delaunay, I decided to call the `InitializeTriangulateIO` as it is plus this extra lines to avoid duplicating this sensible part of the code.

> > In any case I don't think it is a bad design at all as we are just simply "hiding" de allocation and deallocation instructions in two auxiliary functions.
> 
> this makes perfect sense but it is not clear from reading the code! Hence my comments/confusion. I see `new` or `malloc` but not the corresponding `delete`/`free`. Also the function name doesn't make this clear.
> IMO you could improve the naming of the methods to make clear that they do sth with memory as this is a sensitive thing
> 
> In any case I am only suggesting, maybe I don't see all the points
> 

If @loumalouomega agrees (he was the main contributor of this) I'm fine about refactoring the `InitializeTriangulateIO` and `CleanTriangulateIO` to make this more self descriptive.

> Only once I activate valgrind in the CI I might come back and knock at your door ;) :D

:fearful: :running_man:

Reply:
(in any case I improved a bit the documentation to explain what's happening in there)

Reply:
I am OK

Reply:
Then, I suggest to merge this to then do a PR with refactoring changes.

Reply:
Indeed